mock/py/mockbuild/plugins/root_cache.py

```diff
@@ -127,6 +129,35 @@ def _unpack_root_cache(self):
         if self.rootCacheLock is None:
             self.rootCacheLock = open(os.path.join(self.rootSharedCachePath, "rootcache.lock"), "a+")
 
+        if self.buildroot.is_bootstrap and self.buildroot.use_bootstrap_image \
+                and not os.path.exists(self.rootCacheFile):
+            getLog().info("Using bootstrap image: %s" % self.buildroot.bootstrap_image)
+
+            # pull the latest image
+            getLog().info("Pulling image: %s" % self.buildroot.bootstrap_image)
+            cmd = ["podman", "pull", self.buildroot.bootstrap_image]
+            do(cmd, printOutput=True)
+
+            # start a container and detach immediately
+            cmd = ["podman", "run", "--detach", self.buildroot.bootstrap_image, "/bin/true"]
+            container_id = do(cmd, returnOutput=True)
+            container_id = container_id.strip()
+
+            # export container and compress it
+            getLog().info("Exporting container: %s as %s" % (self.buildroot.bootstrap_image, self.rootCacheFile))
+            cmd_podman = ["podman", "export", container_id]
+            podman = subprocess.Popen(cmd_podman, stdout=subprocess.PIPE)
+            cache_file = open(self.rootCacheFile, "w")
+            cmd_compressor = [self.compressProgram, "--stdout"]
+            compressor = subprocess.Popen(cmd_compressor, stdin=podman.stdout, stdout=cache_file)
+            compressor.communicate()
+            podman.communicate()
```

Comment:
Are you sure this will work? I was under the impression, that communicate() cannot be used for large data and you have to use select().

Reply:
I believe this mimics pipe in shell.
Write blocks on full buffer, read blocks on reading from an empty buffer.

AFAIK select() is necessary if you need non-blocking operations.